### PR TITLE
ECE120 Course Offering Page

### PR DESCRIPTION
**Pull request by upai02**
Also made some minor edits to the curriculum pages

### CODE REVIEW

File: docs/Course Wiki/ECE Course Offerings/ECE120.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The course starts by covering the many different representations of information in the form of bits in a computing system, then moves on to cover Boolean algebra as it relates to digital networks, combinational and sequential circuits, finite state machines and memories, and ends with a basic overview of computer organization and assembly programming principles using a basic educational ISA known as LC-3. Some C programming is also covered in the first few weeks through a series of small labs, but no prior programming experience is needed and mastery of C is not expected by the end of this course, as [ECE220](ECE220.md) (the next course in line after ECE120) starts from where this course leaves off and covers C in much more detail. 
+
+## Prerequisites
+
+There are no technical prerequisites to take ECE120 as it is an introductory course to computing fundamentals, and aims to provide an overview of all the content encapsulated by the field, which future classes will expand on by diving deeper into specific topics, such as low level programming (ECE220), [ECE391](ECE391.md)), digital design ([ECE385](ECE385.md)), and computer organization ([ECE411](ECE411.md)).
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: this sentence is pretty long, can we split it

File: docs/Course Wiki/ECE Course Offerings/ECE120.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+NOTE: While electrical engineering students will be better suited to take ECE120 in the spring to make room early on for [ECE110](ECE110.md) and their other classes, it should be taken into consideration that [ECE210](ECE210.md) (the follow-up class for ECE110) has more prerequisites than ECE220. Thus, an electrical engineer may find it a better option to take ECE120 their first semester anyways, just so they can double up on ECE110 and ECE220 while they complete the prerequisites for ECE210. Of course, ECE210 is a fundamental EE course and should be taken as soon as possible, so plan accordingly. For a department-provided suggested schedule, refer to the [EE Curriculum](../../Curriculum%20Graph/1.EE%20Curriculum.md) section.
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: Can this be more concise?

File: docs/Course Wiki/ECE Course Offerings/ECE120.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Resources
+
+Apart from the class wiki page, the [Introduction to Computing Systems: from Bits & Gates to C & Beyond](http://highered.mheducation.com/sites/0072467509/index.html) (2nd edition) textbook written by Patt and Patel offers an expansive detail of the coursework and includes some homework problems that additionally improve understanding and preparation for testing. Course notes, prepared by Professor Lumetta, are available at the ECE Supply Center and [online](https://wiki.illinois.edu/wiki/download/attachments/728159173/ece120-fall-2020-notes-for-students.pdf?version=1&modificationDate=1598033876080&). Other available resources include office hours for material mastery and clarification, and an open lab that allows students to finish lab work.
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: specifically 2nd edition?

**Review by rzig — CHANGES_REQUESTED** [review]
minor writing nits, someone else who has taken the course can comment on content

File: docs/Course Wiki/ECE Course Offerings/ECE120.md (hunk 3 above)
Comment (upai02) [reply docs/Course Wiki/ECE Course Offerings/ECE120.md]: I took that part from the old course wiki, which listed it as such, so I kept it in.

File: docs/Course Wiki/ECE Course Offerings/ECE120.md (hunk 1 above)
Comment (upai02) [reply docs/Course Wiki/ECE Course Offerings/ECE120.md]: Split the sentence into two, should be resolved on newest commit.

File: docs/Course Wiki/ECE Course Offerings/ECE120.md (hunk 2 above)
Comment (upai02) [reply docs/Course Wiki/ECE Course Offerings/ECE120.md]: I have made it a bit more concise on the latest commit, but if needed I can try and cut down more.

**Comment by upai02** [comment]
Requested changes have been made for now

File: docs/Course Wiki/ECE Course Offerings/ECE120.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Infamous Topics
+
+- Sequential logic design: This is covered about halfway through the course, and alongside combinational design forms the backbone of every computing device imaginable. The idea of sequential design centers around feeding combinational gate outputs back into each other and observing changes over time with a varying clock. This can be a bit obtuse to understand, but is vital to know for designing finite state machines, memories, and eventually a full CPU, which the class builds up to in the coming weeks.
+- Finite state machines: Covered soon after sequential logic design, finite state machines (FSMs) are the core of a computing device as they dictate what decision to make given a current state of operation and inputs. Finite state machine design is often diffucult for students to understand and apply to given scenarios, made even harder if they don't properly understand sequential logic, which is what FSMs build upon. This is why, for good reason, many students consider finite state machines to be one of the hardest, if not the hardest part of the course.
```
Comment (lnguy7) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: We may wanna add serialization and IEEE-754 Floating Point bc I feel like students struggle with that

**Review by lnguy7 — COMMENTED** [review]
Other than perhaps this LGTM @alex-m-zhang @upai02

File: docs/Course Wiki/ECE Course Offerings/ECE120.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+There is an additional "Honors Lab" students can take to gain James Scholar credit, newly officially listed this semester (Fall 2023) as [ECE198HL]. This course allows students to design and build any project they want (with TA approval). The department will fund any project in this section (within reason). James Scholars are highly encouraged to take this course to obtain honors credit, but even non-James Scholars may take it to gain more project-based experience early on in their academic careers.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: Add the markdown link to ECE198HL.

File: docs/Course Wiki/ECE Course Offerings/ECE120.md (hunk 4 above)
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE120.md]: "Finite state machine design is often difficult for students to understand" (difficult is spelt wrong).
Also, is FSM design also about algorithm design in some sense? Might be worth mentioning, as that can be difficult.

**Review by alex-m-zhang — CHANGES_REQUESTED** [review]
Resolve the merge conflicts by undoing your changes in the EE Curriculum.md and CE Curriculum.md pages. Those pages are already updated to look proper (and have everything spelled properly) in the main branch.

Otherwise, LGTM except for the comments below. :)

File: docs/Course Wiki/ECE Course Offerings/ECE120.md (hunk 5 above)
Comment (upai02) [reply docs/Course Wiki/ECE Course Offerings/ECE120.md]: Is that link up yet? Didn't realize it was, I'll add it in.

**Comment by upai02** [comment]
Closed for now to get the curriculum page changes merged, will open another PR instead

**Pull request closed without merging**